src/query/api/v1/handler/prometheus/native/read.go

```diff
@@ -75,10 +77,12 @@ type blockWithMeta struct {
 func NewPromReadHandler(
 	engine *executor.Engine,
 	tagOpts models.TagOptions,
-) http.Handler {
+	limitsCfg *config.LimitsConfiguration,
```

Comment:
nit: just wondering why this is a pointer, I think we usually try to keep configs immutable

Reply:
Ahh yeah, this is one of my pet peeves with golang--the guidance on when to use pointers vs not is unclear imo. My intent with using a pointer here isn't to allow mutation (totally agreed with leaving configs immutable after construction), but more to prevent unnecessary copying. Now, you could very reasonably argue that this struct is tiny, making copying quite cheap, and you'd be correct. However, we'll almost certainly be adding fields here, and I in general don't know when the size cutoff makes it better to pass structs by value instead of by reference. As a result, I tend to pass most things by reference unless they're explicitly going to remain small (`Datapoint` is a good example of this). 

Very welcome to other arguments on this, and sorry for the longish response! This is definitely something I'd love more clear guidance on :)

Reply:
Sure, no worries! I think in situations like this where we only use it once on server startup it's ok to copy the struct to get immutability

I think configs in general we copy where possible